### PR TITLE
Add test suite that only requires HMS

### PR DESCRIPTION
**Pull request by homar**
Test Hive and Iceberg connectors with vanilla Apache Hive 3.1.2
(i.e. without HDP's modifications).

Since HiveServer2 startup in currently unreliable in hive3.1-hive testing image,
this includes only tests that do not require HiveServer2.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,369 +26,15 @@ concurrency:
...
-      - name: Upload test results
-        uses: actions/upload-artifact@v2
-        if: failure()
-        with:
-          name: result test-memsql
-          path: |
-            **/target/surefire-reports
-            **/target/checkstyle-*
-
-  test-bigquery:
```
Comment (findepi) [inline .github/workflows/ci.yml:339]: remember to back out

File: plugin/trino-hive-hadoop2/conf/hive-tests-config-hms.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export HADOOP_BASE_IMAGE="ghcr.io/trinodb/testing/hive3.1-hive"
```
Comment (findepi) [inline plugin/trino-hive-hadoop2/conf/hive-tests-config-hms.sh:1]: call this file still `hive-tests-config-apache-hive3.sh`

File: plugin/trino-hive-hadoop2/conf/hive-tests-config-hms.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+export TEMPTO_ENVIRONMENT_CONFIG_FILE="/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms.yaml"
```
Comment (findepi) [inline plugin/trino-hive-hadoop2/conf/hive-tests-config-hms.sh:2]: `tempto-configuration-for-hms-only.yaml`

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/common/standard/trino-init-hive3.sh [hunk 4]
```diff
@@ -3,7 +3,7 @@
 set -xeuo pipefail
 
 for hive_properties in $(find '/docker/presto-product-tests/conf/presto/etc/catalog' -name '*hive*.properties'); do
-    echo "Updating $hive_properties for HDP3"
+    echo "Updating $hive_properties for Hive3"
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/common/standard/trino-init-hive3.sh:6]: Changes here (file rename, comment change) are no longer related to the actual commit, but still good thing to do.
Let's make them into separate commit (or PR).

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/configs/ConfigHms.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import io.trino.tests.product.launcher.docker.DockerFiles;
+import io.trino.tests.product.launcher.env.Environment;
+
+import java.util.List;
+
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.PRESTO;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+public class ConfigHms
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/configs/ConfigHms.java:27]: `ConfigApacheHive3` (i just realized adding "Apache" might be helpful)

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/configs/ConfigHms.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                        dockerFiles.getDockerFilesHostPath("common/standard/trino-init-hive3.sh")),
+                        "/docker/presto-init.d/trino-init-hive3.sh");
+            }
+        });
+    }
+
+    @Override
+    public List<String> getExcludedGroups()
+    {
+        return ImmutableList.of("hive_transactional", "skip_on_cdh");
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/configs/ConfigHms.java:61]: Why skip_on_cdh ?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/configs/ConfigHms.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainers(container -> {
+            if (container.getLogicalName().startsWith(PRESTO)) {
+                container.withCopyFileToContainer(forHostPath(
+                        // Hive 3.1's handling of timestamps is incompatible with previous versions of Hive (see https://issues.apache.org/jira/browse/HIVE-21002);
+                        // in order for Presto to deal with the differences, we must set catalog properties for Parquet and RCFile
+                        dockerFiles.getDockerFilesHostPath("common/standard/trino-init-hive3.sh")),
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/configs/ConfigHms.java:52]: The HMS-only tests (as currently run on this env), should not require this.

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+hdfs:
+    webhdfs:
+        # 9870 is the name node's default port in Hadoop 3
+        uri: http://${databases.hive.host}:9870
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms.yaml:4]: instead of defining it here, let's make the new config load both: `tempto-configuration-for-hive3` and also `tempto-configuration-for-hms-only` -- new file with just 
```
hive:
    # to make sure hive server is not used, we only want to run tests that uses HMS
    jdbc_url: jdbc:hive2://${databases.hive.host}:99999
```

**Comment by homar** [comment]
@findepi could you please take a look again ?;)

File: plugin/trino-hive-hadoop2/conf/hive-tests-config-apache-hive3.sh [hunk 9]
```diff
@@ -0,0 +1,2 @@
+export HADOOP_BASE_IMAGE="ghcr.io/trinodb/testing/hive3.1-hive"
```
Comment (findepi) [inline plugin/trino-hive-hadoop2/conf/hive-tests-config-apache-hive3.sh:1]: This file is currently unused. Add something like https://github.com/trinodb/trino/blob/d3a6c2a91d8f786d263ac846245b7f8de5b752a6/.github/workflows/ci.yml#L119 and let's have an issue for that.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/configs/ConfigApacheHive3.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    @Override
+    public String getHadoopBaseImage()
+    {
+        return "ghcr.io/trinodb/testing/hive3.1-hive";
+    }
+
+    @Override
+    public List<String> getExcludedGroups()
+    {
+        return ImmutableList.of("hive_transactional");
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/configs/ConfigApacheHive3.java:32]: we can remove this for now, since we don't run any HS2-dependent tests anyway yet.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteHmsOnly.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+public class SuiteHmsOnly
+        extends Suite
+{
+    @Override
+    public List<SuiteTestRun> getTestRuns(EnvironmentConfig config)
+    {
+        return ImmutableList.of(
+                testOnEnvironment(EnvSinglenode.class)
+                        .withGroups("hms_only", "hdfs_no_impersonation")
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteHmsOnly.java:34]: any reason hdfs_no_impersonation is here?

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml [hunk 12]
```diff
@@ -0,0 +1,7 @@
+databases:
+    hive:
+        jdbc_url: jdbc:hive2://${databases.hive.host}:12345
+        prepare_statement:
+            - USE ${databases.hive.schema}
+            # Hive 3 gathers stats by default. For test purposes we need to disable this behavior.
+            - SET hive.stats.column.autogather=false
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml:7]: redundant, remove

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml [hunk 13]
```diff
@@ -0,0 +1,7 @@
+databases:
+    hive:
+        jdbc_url: jdbc:hive2://${databases.hive.host}:12345
```
Comment (findepi) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml:3]: add a comment why 12345, and what's the intent of all this

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java [hunk 14]
```diff
@@ -391,7 +392,7 @@ public void testInsertAndSelectWithNullFormat(StorageFormat storageFormat)
 
         assertThat(query(format("SELECT * FROM %s", tableName))).containsOnly(storedValues);
 
-        onHive().executeQuery(format("DROP TABLE %s", tableName));
+        query(format("DROP TABLE %s", tableName));
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:395]: The change here is correct, but should be separate commit, titled something like

```
Drop table using same executor it was created with
```

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHmsOnly.java [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.HMS_ONLY;
+import static io.trino.tests.product.hive.HiveProductTest.ERROR_COMMITTING_WRITE_TO_HIVE_ISSUE;
+import static io.trino.tests.product.hive.HiveProductTest.ERROR_COMMITTING_WRITE_TO_HIVE_MATCH;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+
+public class TestHmsOnly
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHmsOnly.java:27]: Does it add more value as compared to hms_only methods in TestHiveStorageFormats?

**Review by findepi — COMMENTED** [review]
LGTM

Please rename (commit and PR) to something like

```
Test Hive and Iceberg with Apache Hive 3.1.2

Test Hive and Iceberg connectors with vanilla Apache Hive 3.1.2
(i.e. without HDP's modifications).

Since HiveServer2 startup in currently unreliable in hive3.1-hive testing image,
this includes only tests that do not require HiveServer2.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java (hunk 14 above)
Comment (homar) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java:395]: without this change this particular test cannot be run in HMS_ONLY suite, this is why I think it suits in this commit

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHmsOnly.java (hunk 15 above)
Comment (homar) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHmsOnly.java:27]: These 2 tests here are very similar to tests with the same names in `TestHiveCreateTable` however those tests run some checks `onHive` which can't be supported here.  However now I see that without those checks they don't add much more value so I will remove them, thanks

**Comment by homar** [comment]
@findepi  could you please take another look ?

File: .github/workflows/ci.yml [hunk 16]
```diff
@@ -434,6 +435,9 @@ jobs:
           # this suite is not meant to be run with different configs
           - config: default
             suite: suite-compatibility
+          # this suite is not meant to be run with different configs
+          - config: apache-hive3
+            suite: suite-hms-only
```
Comment (findepi) [inline .github/workflows/ci.yml:440]: change comment to
```
# this suite is designed specifically for apache-hive3. TODO remove the suite once we can 
# all regular tests on apache-hive3.

File: .github/workflows/ci.yml [hunk 17]
```diff
@@ -117,6 +117,7 @@ jobs:
           - config-empty
           - config-hdp3
           # TODO: config-cdh5
+          # TODO: config-apache-hive3
```
Comment (findepi) [inline .github/workflows/ci.yml:120]: add similar TODO in `pt` job's `matrix.config`

File: plugin/trino-hive-hadoop2/conf/hive-tests-config-apache-hive3.sh [hunk 18]
```diff
@@ -0,0 +1,2 @@
...
+export TEMPTO_ENVIRONMENT_CONFIG_FILE="/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml"
```
Comment (findepi) [inline plugin/trino-hive-hadoop2/conf/hive-tests-config-apache-hive3.sh:2]: we likely should load both:

```
/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hive3.yaml,/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml
```

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml (hunk 13 above)
Comment (findepi) [reply testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml:4]: maybe
```
# Make hive server configuration invalid to make sure the hms_only tests do not accidentally depend on HS2.
```

**Review by findepi — APPROVED** [review]
Please rebase.

**Comment by homar** [comment]
@findepi  comments addressed, tests seem to work

**Comment by findepi** [comment]
CI https://github.com/trinodb/trino/issues/9263

File: plugin/trino-hive-hadoop2/conf/hive-tests-config-apache-hive3.sh [hunk 19]
```diff
@@ -0,0 +1,3 @@
+export HADOOP_BASE_IMAGE="ghcr.io/trinodb/testing/hive3.1-hive"
+export TEMPTO_ENVIRONMENT_CONFIG_FILE="/docker/presto-product-tests/conf/tempto/tempto-configuration-for-hms-only.yaml
+"
```
Comment (findepi) [inline plugin/trino-hive-hadoop2/conf/hive-tests-config-apache-hive3.sh:3]: stray newline

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
Merged, thanks!
